app/services/search/vacancy_search.rb

```diff
@@ -50,18 +49,35 @@ def vacancies
   end
 
   def total_count
-    @total_count ||= vacancies.count
+    @total_count ||= vacancies.count(:id)
   end
 
   private
 
   def scope
+    sort_by_distance = sort.by == "distance"
     scope = Vacancy.live.includes(:organisations)
     scope = scope.where(id: organisation.all_vacancies.pluck(:id)) if organisation
-    scope = scope.search_by_location(location, radius) if location
+    scope = scope.search_by_location(location, radius, sort_by_distance: sort_by_distance) if location
     scope = scope.search_by_filter(search_criteria) if search_criteria.any?
     scope = scope.search_by_full_text(keyword) if keyword.present?
-    scope = scope.reorder(sort.by => sort.order) if sort&.by_db_column?
-    scope
+    order_scope(scope, sort_by_distance)
+  end
+
+  def sort_by
+    if sort.by == "publish_on_non_default"
+      "publish_on"
+    else
+      sort.by
+    end
+  end
+
+  def order_scope(scope, sort_by_distance)
+    # if sort_by_distance is true then the sorting is handled by the search_by_filter method so we do not re-order here.
+    return scope if sort_by_distance
```

Comment:
Suggestion: Given that we already have a guard clause checking for the `sort.by` value, I would personally find this more readable if we avoid passing a second parameter and use the `sort` object directly. But this is just how I personally feel about it, so is not a request!
```suggestion
  def order_scope(scope)
    # if sort_by_distance is true then the sorting is handled by the search_by_filter method so we do not re-order here.
    return scope if sort&.by == "distance"
```